### PR TITLE
[Publisher][Design Update][Metric Settings] Update layout of Metric Settings overview page (1/3)

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

Updates the layout of Metric Settings overview page to match the [Figma designs](https://www.figma.com/file/dvX3yg7FMGJSWFaqHWrWgQ/Justice-Counts-Publisher?type=design&node-id=9418%3A4011&mode=design&t=6EO7qWlRKq8bn1bC-1).

The changes are purely styling changes - no functionality was touched/updated.

Note: there's a new treatment for handling an agency w/ a long list of sectors in the overview page incoming - will update this PR once that design is ready.

Note: I'm also going to hold off merging these updates until parts 1, 2, and 3 are ready to go at once and we're ready to release.

Demo: https://mahmoudtest---publisher-web-b47yvyxs3q-uc.a.run.app/

## Related issues

Contributes to #1167 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: publisher/src/components/MetricsConfiguration/MetricsOverview.tsx [hunk 1]
```diff
@@ -108,98 +121,88 @@ export const MetricsOverview = observer(() => {
...
-                      selected={currentSystem === system}
-                      onClick={() => handleSystemClick(system)}
-                    >
-                      {formatSystemName(system, {
-                        allUserSystems: currentAgency?.systems,
-                      })}
-                    </Styled.SystemMenuItem>
-                  );
-                })}
-          </Styled.SystemsList>
```
Comment (mxosman) [inline publisher/src/components/MetricsConfiguration/MetricsOverview.tsx:131]: Converted this to the TabbedBar structure (see L89-99, and L129)

File: publisher/src/components/MetricsConfiguration/MetricsOverview.tsx [hunk 2]
```diff
@@ -108,98 +121,88 @@ export const MetricsOverview = observer(() => {
...
+                  <Styled.MetricItem
+                    key={key}
+                    onClick={() =>
+                      setSettingsSearchParams({
+                        system: currentSystem,
+                        metric: key,
+                      })
+                    }
+                  >
+                    <Styled.MetricItemName actionRequired>
```
Comment (mxosman) [inline publisher/src/components/MetricsConfiguration/MetricsOverview.tsx:147]: Added an `actionRequired` flag here for styling

File: publisher/src/components/MetricsConfiguration/MetricsOverview.tsx [hunk 3]
```diff
@@ -108,98 +121,88 @@ export const MetricsOverview = observer(() => {
...
-                })}
-          </Styled.SystemsList>
+
+          {/* System Selection */}
+          {showSystems && <TabbedBar options={systemTabOptions} />}
+
+          <Styled.MetricsWrapper>
+            {hasActionRequiredMetrics && (
+              <Styled.MetricsSection>
+                <Styled.MetricsSectionTitle isAlertCaption>
```
Comment (mxosman) [inline publisher/src/components/MetricsConfiguration/MetricsOverview.tsx:134]: Added an `isAlertCaption` flag here for styling

File: publisher/src/components/MetricsConfiguration/MetricsOverview.tsx [hunk 4]
```diff
@@ -108,98 +121,88 @@ export const MetricsOverview = observer(() => {
             Click on each metric to edit the availability of the metric and
             relevant breakdown categories, as well as the definitions of each.{" "}
             <a href={learnMoreURL} target="_blank" rel="noopener noreferrer">
-              Learn More
+              Learn more
```
Comment (morden35) [inline publisher/src/components/MetricsConfiguration/MetricsOverview.tsx:124]: nice catch!

**Review by morden35 — APPROVED** [review]
Looks great Mahmoud! I would note that the font size looks a bit small to me (at least compared to what we had previously / the home page). I would also note that we may want to eventually update the help center screenshots once this is all done (obviously non-blocking).

**Comment by lilidworkin** [comment]
Looks SO GOOD!!!! SUCH a huge improvement

re: font size, yea, what I noticed especially was how the "Available" and "Not Available" headers felt easy to miss. Otherwise I actually don't mind the smaller sizes

**Comment by lilidworkin** [comment]
Two other VEFY small nits, maybe just file away to discuss with @forrestkim00 at some point?

Looks like an indent on the race/ethnicity form:
Screenshot 2024-02-14 at 2 07 40 PM

The tooltip font now feels too large / unbalanced?
Screenshot 2024-02-14 at 2 08 20 PM

**Comment by mxosman** [comment]
Thank you, Michelle & Lili!

> Looks great Mahmoud! I would note that the font size looks a bit small to me (at least compared to what we had previously / the home page).

>re: font size, yea, what I noticed especially was how the "Available" and "Not Available" headers felt easy to miss. Otherwise I actually don't mind the smaller sizes

Yeah - it took me a little bit for my eyes to adjust, especially comparing the current settings page, but it's oddly enough normalized for me now. It's at 14px vs. before it was I think around 16px. 

I personally don't mind the font size, but completely understand if folks feel like things will be easy to miss - looping in @forrestkim00 for his thoughts as well.

> I would also note that we may want to eventually update the help center screenshots once this is all done (obviously non-blocking).

Yes, great call! This is on my personal roadmap for this project!

>Looks like an indent on the race/ethnicity form:

Oh that's my bad - there should be an X close button there on that row! Great catch!

>The tooltip font now feels too large / unbalanced?

The body font is the same size as the metric name - but let me update the header font and see if that normalizes things.

**Comment by forrestkim00** [comment]
Yep, you're correct the font got smaller @morden35 ! I think once we make the typogrpahic changes to the other pages, it should look consistent. 

@lilidworkin I can play around with "Available" and "Not Available" visibility. In the meantime, I made a quick [prototype](https://www.figma.com/proto/dvX3yg7FMGJSWFaqHWrWgQ/Justice-Counts-Publisher?page-id=8917%3A1478&type=design&node-id=10458-9965&viewport=-66693%2C-17851%2C0.72&t=b2XIFnC8kn5xG4rc-1&scaling=min-zoom&starting-point-node-id=10458%3A9965&mode=design) of the tooltip. I made the font a tad smaller and centered the tooltip to the metric. LMK if this is moving in the right direction!

**Comment by mxosman** [comment]
> Yep, you're correct the font got smaller @morden35 ! I think once we make the typogrpahic updates to the other pages, it should look consistent.
> 
> @lilidworkin I can play around with "Available" and "Not Available" visibility. In the meantime, I made a quick [prototype](https://www.figma.com/proto/dvX3yg7FMGJSWFaqHWrWgQ/Justice-Counts-Publisher?page-id=8917%3A1478&type=design&node-id=10458-9965&viewport=-66693%2C-17851%2C0.72&t=b2XIFnC8kn5xG4rc-1&scaling=min-zoom&starting-point-node-id=10458%3A9965&mode=design) of the tooltip. I made the font a tad smaller and centered the tooltip to the metric. LMK if this is moving in the right direction!

TY for your thoughts & quick exploration, Forrest! Quick question - how can I access the original Figma nodes from the prototype to check the values of the font?

Also, since we're using React Tooltip for the tooltips and the placement of it in the center like you have it in the prototype is not one of the options it has - any chance you'd be open to one of the available positions here: https://react-tooltip.com/docs/examples/place? I could try and create a custom override - but would rather not if possible as that can get tricky.

**Comment by forrestkim00** [comment]
Yes, I can drop it in to the handoff file and @ you! 

I had one quick question about the React Tooltip. Right now is the value top-end? If it were just top would it be in the middle on top of the component, and if it were top-start it would be on top of the component but aligned left?

**Comment by mxosman** [comment]
> Yes, I can drop it in to the handoff file and @ you!
> 
> I had one quick question about the React Tooltip. Right now is the value top-end? If it were just top would it be in the middle on top of the component, and if it were top-start it would be on top of the component but aligned left?

Sorry I didn't mention before - yes, you're absolutely right - it's `top-end`. And you're spot on about the `top` and `top-start`.

`top`:
Screenshot 2024-02-15 at 4 59 27 PM

`top-start`:
Screenshot 2024-02-15 at 5 00 07 PM

**Comment by mxosman** [comment]
WAIT - Forrest, I just found an `offset` property within ReactTooltips that will accomplish this! This is possible within the library itself - sorry about that! Just caught it now playing around with it... Please disregard! 😓 

Screenshot 2024-02-15 at 5 02 13 PM

**Comment by mxosman** [comment]
Hmm - though, the centering does seem to block some of the longer names:

Screenshot 2024-02-15 at 5 06 02 PM

**Comment by mxosman** [comment]
Hi @lilidworkin & @forrestkim00 -- 

So I went ahead and updated the `font-size` of the tooltip title to match the body, and reduced the tooltip content font (which was already the body size) by 1px to 13px - here is how it looks now:

Screenshot 2024-02-20 at 10 16 05 AM

I also kept the tooltip showing up at `top-end` to avoid it overlapping with long definition names.

Let me know if this works all around! I'm redeploying to playtesting now - so it should be live in about ~11 min or so.

**Comment by forrestkim00** [comment]
I think this looks great, I think the smaller font makes it look a lot better. Thanks Mahmoud! Sorry about the delayed reply!

**Comment by forrestkim00** [comment]
Also, referring to @lilidworkin comment about the missable "Available", "Unavailable" headers. I made a quick redesign of it. IDK if we want to address it now or prioritize it later? 
image

**Comment by mxosman** [comment]
> I think this looks great, I think the smaller font makes it look a lot better. Thanks Mahmoud! Sorry about the delayed reply!

Awesome! Thank you, @forrestkim00! No apologies necessary at all - I won't merge these PRs in until I get the supervision subpop + guides updated anyway.

I'm realizing that the tooltip covering the long definition names doesn't matter because the name is on the tooltip itself 😓 - sorry about that, Forrest! I'll move the tooltip back to the center as you had it in the screenshot because it actually feels much more clear that it's tied to the metric hovered compared to the `top-end` position.

>Also, referring to @lilidworkin comment about the missable "Available", "Unavailable" headers. I made a quick redesign of it. IDK if we want to address it now or prioritize it later?

I'm down to address this now within these PRs!

**Pull request merged**